### DISCUSSION TITLE
Nie blokować zmiany stanów magazynowych w wystawionych ofertach po zmianach w produktyzacji

### DISCUSSION THREAD

**Discussion by alekskuc** [allegro/allegro-api#5376] [Suggestions - resolved]
Ogólny scenariusz:
1. Nie blokować zmiany stanów magazynowych (/sale/offer-quantity-change-commands/ błąd MissingRequiredParameters) w wystawionych ofertach po dodaniu nowych wymaganych parametrów w powiązanym produkcie z katalogu allegro.
2.  Dodany nowy wymagany parametr w produkcie skopiować do wystawionych ofert.

Szczegółowy scenariusz:
1. Wystawiam ofertę bez parametru Marka, bo aktualnie nie istnieje w tej kategorii.
2. Wiąże ofertę z produktem z produktyzacji i wszystkie wymagane parametry się przepisują z produktu, co akceptuję.
3. Powiązany produkt w produktyzacji zostaje zmieniony i dochodzi nowy parametr Marka oraz zostają zmienione inne parametry.
4. W przypadku próby zmiany stanów magazynowych jest aktualnie błąd braku parametru Marka, który nie istniał w momencie wystawiania oferty w powiązanym produkcie. O tym, że inne się różnią, nie ma błędu, bo są uzupełnione. W tym miejscu proponuję zmianę żeby wyłącznie nieistniejące parametry takie jak Marka dodać do oferty bez potwierdzania, bo potwierdzenie było w momencie powiązania z produktem przy wystawianiu. Samo uzupełnienie oferty o wymagany nieistniejący uprzednio parametr Marka nie nadpisuje danych które by mógł ustawić sprzedawca, a zapewnia ciągłość sprzedaży.

W nawiązaniu do #5361

**Comment by aleksandrasami** [comment]
Hej,
dołączam się do powyższej sugestii. Scenariusz pozwala uniknąć dodatkowych trudności w związku z zarządzaniem swoimi ofertami i tak jak wspomina @alekskuc zapewniłby ciągłość sprzedaży. Według mnie jest to rozwiązanie przyjazne dla użytkownika:)

**Comment by MartaNowaczyk** [comment]
Rozwiązaniem jest metoda [PATCH /sale/product-offers/{offerId}](https://developer.allegro.pl/documentation/#operation/editProductOffers), gdy przesyłasz/edytujesz tylko stan magazynowy. Sugestię oznaczam jako wdrożoną.
